src/pool.jl

```diff
@@ -3,13 +3,17 @@ function CategoricalPool(index::Vector{S},
                          order::Vector{R},
                          ordered::Bool=false) where {S, T <: Integer, R <: Integer}
     invindex = convert(Dict{S, R}, invindex)
-    CategoricalPool{S, R, CategoricalValue{S, R}}(index, invindex, order, ordered)
+    C = catvaluetype(S, R)
+    V = leveltype(C) # might be different from S (e.g. S == SubString, V == String)
+    CategoricalPool{V, R, C}(index, invindex, order, ordered)
 end
 
+CategoricalPool{T, R, C}(ordered::Bool=false) where {T, R, C} =
+    CategoricalPool{T, R, C}(T[], Dict{T, R}(), R[], ordered)
 CategoricalPool{T, R}(ordered::Bool=false) where {T, R} =
     CategoricalPool(T[], Dict{T, R}(), R[], ordered)
```

Comment:
Why not change this function just like the one below?

Reply:
The one above calls inner ctor directly since `C` is explicitly specified.
The one below calls the one in question.
The one in question calls the first ctor in `pool.jl`, which handles `C` selection.
So I have to repeat `T[], Dict{T,R}...` because they are passed to 2 different ctors.

Reply:
Actually, I would have kept the method below as it was.